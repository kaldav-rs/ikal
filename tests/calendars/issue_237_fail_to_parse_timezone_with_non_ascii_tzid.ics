BEGIN:VCALENDAR
VERSION:2.0
PRODID:collective/icalendar
BEGIN:VTIMEZONE
TZID:(UTC-03:00) Brasília
BEGIN:STANDARD
TZNAME:Brasília standard
DTSTART:16010101T235959
TZOFFSETFROM:-0200
TZOFFSETTO:-0300
RRULE:FREQ=YEARLY;INTERVAL=1;BYDAY=3SA;BYMONTH=2
END:STANDARD
BEGIN:DAYLIGHT
TZNAME:Brasília daylight
DTSTART:16010101T235959
TZOFFSETFROM:-0300
TZOFFSETTO:-0200
RRULE:FREQ=YEARLY;INTERVAL=1;BYDAY=2SA;BYMONTH=10
END:DAYLIGHT
END:VTIMEZONE
BEGIN:VEVENT
DTSTART;TZID="(UTC-03:00) Brasília":20170511T133000
DTEND;TZID="(UTC-03:00) Brasília":20170511T140000
DTSTAMP:16010101T235959
UID:123456
END:VEVENT
END:VCALENDAR
