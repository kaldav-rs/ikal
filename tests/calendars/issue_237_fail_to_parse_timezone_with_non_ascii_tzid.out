Ok(
    VCalendar {
        prodid: Text {
            params: {},
            text: "collective/icalendar",
        },
        version: Text {
            params: {},
            text: "2.0",
        },
        calscale: None,
        method: None,
        alarms: [],
        events: [
            VEvent {
                dtstamp: Naive(
                    1601-01-01T23:59:59,
                ),
                uid: Text {
                    params: {},
                    text: "123456",
                },
                dtstart: DateTime(
                    Naive(
                        2017-05-11T13:30:00,
                    ),
                ),
                class: None,
                created: None,
                description: None,
                geo: None,
                last_modified: None,
                location: None,
                organizer: None,
                priority: None,
                sequence: None,
                status: None,
                summary: None,
                transp: None,
                url: None,
                recurid: None,
                rrule: None,
                dtend: Some(
                    DateTime(
                        Naive(
                            2017-05-11T14:00:00,
                        ),
                    ),
                ),
                duration: None,
                attach: [],
                attendee: [],
                categories: [],
                comment: [],
                contact: [],
                exdate: [],
                rstatus: [],
                related_to: [],
                resources: [],
                rdate: [],
                alarms: [],
                x_prop: {},
                iana_prop: {},
            },
        ],
        freebusy: [],
        journals: [],
        todo: [],
        timezones: [
            VTimezone {
                tzid: Text {
                    params: {},
                    text: "(UTC-03:00) Brasília",
                },
                last_modified: None,
                tzurl: None,
                standard: [
                    Prop {
                        dtstart: DateTime(
                            Naive(
                                1601-01-01T23:59:59,
                            ),
                        ),
                        tzoffsetto: -03:00,
                        tzoffsetfrom: -02:00,
                        rrule: Some(
                            Recur {
                                freq: Yearly,
                                until: None,
                                count: None,
                                interval: 1,
                                by_second: [],
                                by_minute: [],
                                by_hour: [],
                                by_day: [
                                    WeekdayNum {
                                        weekday: Saturday,
                                        ord: Some(
                                            3,
                                        ),
                                    },
                                ],
                                by_monthday: [],
                                by_yearday: [],
                                by_weekno: [],
                                by_month: [
                                    2,
                                ],
                                by_setpos: [],
                                wkst: None,
                            },
                        ),
                        comment: [],
                        rdate: [],
                        tzname: [
                            Text {
                                params: {},
                                text: "Brasília standard",
                            },
                        ],
                        x_prop: {},
                        iana_prop: {},
                    },
                ],
                daylight: [
                    Prop {
                        dtstart: DateTime(
                            Naive(
                                1601-01-01T23:59:59,
                            ),
                        ),
                        tzoffsetto: -02:00,
                        tzoffsetfrom: -03:00,
                        rrule: Some(
                            Recur {
                                freq: Yearly,
                                until: None,
                                count: None,
                                interval: 1,
                                by_second: [],
                                by_minute: [],
                                by_hour: [],
                                by_day: [
                                    WeekdayNum {
                                        weekday: Saturday,
                                        ord: Some(
                                            2,
                                        ),
                                    },
                                ],
                                by_monthday: [],
                                by_yearday: [],
                                by_weekno: [],
                                by_month: [
                                    10,
                                ],
                                by_setpos: [],
                                wkst: None,
                            },
                        ),
                        comment: [],
                        rdate: [],
                        tzname: [
                            Text {
                                params: {},
                                text: "Brasília daylight",
                            },
                        ],
                        x_prop: {},
                        iana_prop: {},
                    },
                ],
                x_prop: {},
                iana_prop: {},
            },
        ],
        x_prop: {},
        iana_prop: {},
    },
)
